nixos/modules/services/web-apps/languagetool.nix

```diff
@@ -0,0 +1,47 @@
+{ config, pkgs, lib, ... }:
+with lib;
+let
+  cfg = config.services.languageToolHttp;
+  format = pkgs.formats.toml { };
+in
+{
+  options.services.languageToolHttp = {
+    enable = mkEnableOption "the LanguagTool http server";
+    allowBrowserPluginAccess = mkEnableOption "access from the browser plugin";
+    settings = mkOption {
+      description = ''
+        Contents of the configuration file.
+      '';
+      inherit (format) type;
+      default = { };
+    };
+    args = mkOption {
+      description = ''
+        Arguments to be passed when starting the server.
+      '';
+      type = types.submodule {
+        freeformType = with types;attrsOf (oneOf [ int str bool ]);
+        options.port = mkOption {
+          description = ''
+            The port which the LanguageTool server listen.
+          '';
+          type = types.port;
+          default = 8081;
+        };
+      };
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.languageToolHttp.args = {
+      allow-origin = mkIf cfg.allowBrowserPluginAccess (mkDefault "*");
+      config = builtins.toString (format.generate "LanguageTool.cfg" cfg.settings);
+    };
+    systemd.services.languagetool-http = {
```

Comment:
This service isn't complete. It doesn't specify a description, user, group, working directory, or a number of other possible values this would benefit from. If you're having trouble you could take inspiration from here: https://github.com/languagetool-org/languagetool/blob/1e740fc3eed858e43ef218bb209b3f055ec0d9d0/languagetool-rpm-package/src/main/resources/templates/languagetool.service.template

Additionally, in NixOS land you get big bonus points for utilizing `systemd` hardening mechanisms. This appears to be a straight forward `java` web application so I'm guessing you could tighten this up pretty hard.